jenkins-scripts/docker/lib/docker_generate_dockerfile.bash

```diff
@@ -33,24 +33,35 @@ export APT_PARAMS=
 
 GZDEV_DIR=${WORKSPACE}/gzdev
 GZDEV_BRANCH=${GZDEV_BRANCH:-master}
+if [[ "$ghprbSourceBranch" =~ ci_matching_branch\/ ]]; then
+  GZDEV_TRY_BRANCH=$ghprbSourceBranch
+fi
 
 dockerfile_install_gzdev_repos()
 {
 cat >> Dockerfile << DELIM_OSRF_REPO_GIT
 RUN rm -fr ${GZDEV_DIR}
-RUN git clone --depth 1 https://github.com/ignition-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
+RUN git clone https://github.com/ignition-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
+RUN if [ -n $GZDEV_TRY_BRANCH ]; then \
+        git -C ${GZDEV_DIR} fetch origin $GZDEV_TRY_BRANCH || true; \
+        git -C ${GZDEV_DIR} checkout $GZDEV_TRY_BRANCH || true; \
+    fi || true"
```

Comment:
The stray quote here was removed via direct push https://github.com/ignition-tooling/release-tools/commit/99524e1347c272920f4b384f3372037d3d38bf36